### PR TITLE
Add delete garden functionality

### PR DESCRIPTION
**Pull request by willkunz13**

### CODE REVIEW

File: app/controllers/user/mygardens_controller.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 class User::MygardensController < User::BaseController 
 	def index
-		@gardens = current_user.gardens
+		@gardens = User.find(current_user.id).gardens
```
Comment (adumortier) [inline app/controllers/user/mygardens_controller.rb:3]: were you running into problems with the way we were accessing the gardens before?

File: spec/features/users/mygarden/delete_garden_spec.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      @garden = Garden.create(name: 'Garden 1', user: @user1)
+      @plant = Plant.new(name: 'tomato', image: "https://www.almanac.com/sites/default/files/image_nodes/tomatoes_helios4eos_gettyimages-edit.jpeg")
+			@garden.plants << @plant
+    end
+
+		it "Delete Garden", :vcr do
+			visit "/user/mygardens"
+			expect(page).to have_content('Garden 1')
+			click_on 'Delete Garden'
+			expect(page).to_not have_content('Garden 1')
```
Comment (adumortier) [inline spec/features/users/mygarden/delete_garden_spec.rb:24]: maybe add that the current path is what we expect(?)
do we want to check that `@user1.gardens` is an empty array?

File: app/controllers/user/mygardens_controller.rb (hunk 1 above)
Comment (willkunz13) [reply app/controllers/user/mygardens_controller.rb:3]: i needs to access the database in order to be truly up to date, especially for the purposes of deleting

File: spec/features/users/mygarden/delete_garden_spec.rb (hunk 2 above)
Comment (willkunz13) [reply spec/features/users/mygarden/delete_garden_spec.rb:24]: that wouldn't be a feature test

**Pull request merged**